File: solutions/fable-1/fable/ruby/primes.rb
n = STDIN.read.to_i
if n < 2
  puts ""
else
  sieve = Array.new(n + 1, true)
  sieve[0] = sieve[1] = false
  i = 2
  while i * i <= n
    if sieve[i]
      j = i * i
      while j <= n
        sieve[j] = false
        j += i
      end
    end
    i += 1
  end
  primes = []
  (2..n).each { |k| primes << k if sieve[k] }
  puts primes.join(" ")
end

File: solutions/fable-1/fable/rust/primes.rs
use std::io::{self, Read, Write};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap_or(0);

    let mut sieve = vec![true; n + 1];
    if n >= 0 {
        sieve[0] = false;
    }
    if n >= 1 {
        sieve[1] = false;
    }
    let mut i = 2;
    while i * i <= n {
        if sieve[i] {
            let mut j = i * i;
            while j <= n {
                sieve[j] = false;
                j += i;
            }
        }
        i += 1;
    }

    let primes: Vec<String> = (2..=n).filter(|&k| sieve[k]).map(|k| k.to_string()).collect();
    let stdout = io::stdout();
    let mut out = stdout.lock();
    writeln!(out, "{}", primes.join(" ")).unwrap();
}
